src/pages/library/datasetBuilder/CohortEditor.ts

```diff
@@ -0,0 +1,429 @@
+      },
+    },
+    [
+      div({ style: { margin: '5px', display: 'flex', alignItems: 'center' } }, [
+        h(
+          Link,
+          {
+            'aria-label': 'delete criteria',
+            onClick: () => {
+              deleteCriteria(criteria);
+            },
+          },
+          [icon('minus-circle', { size: 24, style: { color: colors.danger() } })]
+        ),
+        div({ style: { marginLeft: 5 } }, [
+          Utils.cond(
+            [
+              'category' in criteria,
+              () => {
+                const domainCriteria = criteria as DomainCriteria;
+                return h(Fragment, [strong([`${domainCriteria.category}:`]), ` ${domainCriteria.name}`]);
+              },
+            ],
+            [
+              'value' in criteria,
+              () => {
+                const listCriteria = criteria as ProgramDataListCriteria;
+                return h(Fragment, [strong([`${criteria.name}:`]), ` ${listCriteria.value.name}`]);
+              },
+            ],
+            [
+              'low' in criteria,
+              () => {
+                const rangeCriteria = criteria as ProgramDataRangeCriteria;
+                return h(Fragment, [strong([`${criteria.name}:`]), ` ${rangeCriteria.low} - ${rangeCriteria.high}`]);
+              },
+            ],
+            [Utils.DEFAULT, () => div(['Unknown criteria type'])]
+          ),
+        ]),
+      ]),
+      `Count: ${criteria.count}`,
+    ]
+  );
+
+const selectDomainCriteria = (domainType: DomainType): DomainCriteria => {
+  // This needs to be replaced with a UI that lets users select the criteria they want from
+  // the list of concepts for this domain.
+  return {
+    name: domainType.values[0],
+    id: domainType.id,
+    category: domainType.category,
+    // Need to call the API service to get the count for this criteria.
+    count: 100,
+  };
+};
+
+const createDefaultListCriteria = (listType: ProgramDataListType): ProgramDataListCriteria => {
+  return {
+    name: listType.name,
+    id: listType.id,
+    count: 100,
+    value: listType.values[0],
+  };
+};
+
+const createDefaultRangeCriteria = (rangeType: ProgramDataRangeType): ProgramDataRangeCriteria => {
+  return {
+    name: rangeType.name,
+    id: rangeType.id,
+    count: 100,
+    low: rangeType.min,
+    high: rangeType.max,
+  };
+};
+
+function createCriteriaFromType(
+  type: DomainType | ProgramDataRangeType | ProgramDataListType
+): DomainCriteria | ProgramDataRangeCriteria | ProgramDataListCriteria {
+  return (
+    Utils.condTyped<DomainCriteria | ProgramDataRangeCriteria | ProgramDataListCriteria>(
+      ['category' in type, () => selectDomainCriteria(type as DomainType)],
+      ['values' in type, () => createDefaultListCriteria(type as ProgramDataListType)],
+      ['min' in type, () => createDefaultRangeCriteria(type as ProgramDataRangeType)]
+    ) ?? { category: 'unknown', name: 'unknown', count: 0, id: 0 }
```

Comment:
I seem to remember there being a DEFAULT for cond as well? If there is, we might want to use that rather than `??` here

(It might just be a final function that isn't in an array?)

Reply:
The problem is that `condTyped()` as written can return `undefined`, so I needed the to handle that outside the cond.  And the problem with making `condTyped()` not return `undefined` is that it can't know if it has a `DEFAULT` case or not, so it may or may not have a return type if no case is selected. Since cond isn't a compile time object it also can't tell if it's exhaustive, which is another way `undefined` could be eliminated as a return type.